### DISCUSSION TITLE
ILI9488 with esp32-WROOM-32 parallel

### DISCUSSION THREAD

**Discussion by joachymus** [Bodmer/TFT_eSPI#1761] [Q&A - General]
Dear all,

I am trying to connect esp32-WROOM-32 (38 pins; https://www.theengineeringprojects.com/wp-content/uploads/2020/12/introduction-to-esp32-3.png) with ILI9488 using parallel connection but cannot get it working.
Wroom has different pinout than esp32 and there is an issue... I tried to reconnect multiple ways but still no success. Unfortunately, I have practically no idea what I am doing so it's difficult to troubleshoot. So the only discovery is that:
I have constantly nice white (shining) display with nothing on it. When trying to "mimic" other board connections I reconnect LCD_D7 to TX (pin 35) and LCD_D0 to RX (pin 34) which caused problem with sketch uploading.

I believe that the issue is with the pins although it could be anywhere...

Thanks for any tips!

BR,
j

**Comment by Bodmer** [comment]
The ESP32 pins 34, 35 and 36 are input only, hence the need to connect other pins as detailed in the ReadMe for the UNO shield.

Here is an example setup file detaiing the pins used.
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup14_ILI9341_Parallel.h

**Reply by joachymus** [reply]
Hi Bodmer,
Thank you for your reply.
I am actually not using the above named pins.

This is my board esp32-wroom-32 with 38 pins:

![1649486358654](https://user-images.githubusercontent.com/46739380/162560794-c980589c-1f58-4c82-b8f4-5e667107cc4a.jpg)


First I tried your pin assignment from "Setup16_ILI9488_Parallel":
`#define ESP32_PARALLEL
#define ILI9488_DRIVER

// ESP32 pins used
#define TFT_CS 33 // Chip select control pin
#define TFT_DC 15 // Data Command control pin - must use a pin in the range 0-31
#define TFT_RST 32 // Reset pin

#define TFT_WR 4 // Write strobe control pin - must use a pin in the range 0-31
#define TFT_RD 2

#define TFT_D0 12 // Must use pins in the range 0-31 for the data bus
#define TFT_D1 13 // so a single register write sets/clears all bits
#define TFT_D2 26
#define TFT_D3 25
#define TFT_D4 17
#define TFT_D5 16
#define TFT_D6 27
#define TFT_D7 14`

... it didn't work - white screen
... so I reconnected pin D7 to 11 as 14 is marked as "GND" on my board - white screen
... reconnected RST to 31 as 32 is alo marked as "GND" - same result/white screen.
... reconnectet RD from 2 to 5 as 2 is marked as "EN" - failed to upload sketch - reconnected to several others - fail - back to pin 2
... reconnected WR from 4 to 5 as 4 is marked as "SM"
... reconnected D5 from 16 to 10 as 16 is marked as "SD2"
... reconnected D4 from 17 to 9 as 17 is marked as "SD3

All other pins are marked as "GX" so I guess that those are common GPIOs so there is probably no reason for swapping there and back...

The problem is that I don't know what each display pin needs... and there are to many combinations to simply try everything :(

Any advice would be welcome; Thanks!

**Comment by Bodmer** [comment]
Although in the Arduino world I/O signals get called pins, these are not physical pins but an abstracted name like A0 for an analogue pin. This gives common naming for different processors.

For the ESP32 these pin numbers are actually the GPIO number. So:
#define TFT_D7 14
Means connect the TFT board signal D7 to ESP32 GPIO 14. On your board this is labelled G14.

**Reply by joachymus** [reply]
Hi Bodmer,
That was a solver! It didn't work at the beginning but then I found something about GPIO2 being unable to manage load (or whatever) so I reconnected 2 to 0 and got it show text (Print example).
Now, I am struggling with other examples (cannot upload sketch - "waiting for packet header" - or similar) but at least I know that display works... 
I had no much time to play with it so after I test it, and if still being unable to get it "move" I would raise another query.
For now, I consider this solved and thank you for your help!
BR,
j.

**Reply by Bodmer** [reply]
In stubborn cases press and hold the boards "Boot" button after clickung "Upload", hold until the IDE shows uploading is in progress. If you hold until upload is complete you will need to press the Reset button to start the sketch.
